backend/src/main/java/com/soen341/instagram/service/impl/CommentService.java

```diff
@@ -0,0 +1,161 @@
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Service;
+
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.dao.impl.CommentRepository;
+import com.soen341.instagram.dao.impl.PictureRepository;
+import com.soen341.instagram.exception.comment.CommentLengthTooLongException;
+import com.soen341.instagram.exception.comment.CommentNotFoundException;
+import com.soen341.instagram.exception.like.MultipleLikeException;
+import com.soen341.instagram.exception.picture.InvalidIdException;
+import com.soen341.instagram.exception.picture.PictureNotFoundException;
+import com.soen341.instagram.model.Account;
+import com.soen341.instagram.model.Comment;
+import com.soen341.instagram.model.Picture;
+
+@Service("commentService")
+public class CommentService
+{
+	@Autowired
+	private CommentRepository commentRepository;
+	@Autowired
+	private AccountRepository accountRepository;
+	@Autowired
+	private PictureRepository pictureRepository;
+	@Autowired
+	@Qualifier("UserDetailsService")
+	private UserDetailsService userDetailsService;
+
+	private static int maxCommentLength = 250;
+
+	public Comment createComment(final String commentContent, final long pictureId)
+	{
+		if (commentContent.length() > maxCommentLength)
+		{
+			throw new CommentLengthTooLongException("Comment length exceeds " + maxCommentLength + " characters");
+		}
+
+		final Account account = getCurrentUser();
+		final Optional<Picture> picture = pictureRepository.findById(pictureId);
+
+		if (!picture.isPresent())
+		{
+			throw new PictureNotFoundException();
+		}
+
+		final Comment comment = new Comment();
+		comment.setComment(commentContent);
+		comment.setCreated(new Date());
+		comment.setAccount(account);
+		comment.setPicture(picture.get());
+
+		commentRepository.save(comment);
+		return comment;
+	}
+
+	public int likeComment(final long commentId)
+	{
+		final Comment comment = findComment(commentId);
+		final Set<Account> likedBy = comment.getLikedBy();
+		final boolean addedSuccessfully = likedBy.add(getCurrentUser());
+		if (!addedSuccessfully)
+		{
+			throw new MultipleLikeException("A comment can only be liked one time by the same user");
+		}
+
+		commentRepository.save(comment);
+
+		return likedBy.size();
+	}
+
+	public int unlikeComment(final long commentId)
+	{
+		final Comment comment = findComment(commentId);
+		final Set<Account> likedBy = comment.getLikedBy();
+		final boolean removedSuccessfully = likedBy.remove(getCurrentUser());
+
+		if (!removedSuccessfully)
+		{
+			throw new MultipleLikeException("The comment has not been liked by this user");
```

Comment:
MultipleLikeException does not fit here. We can change this is a future sprint.

Reply:
Why wouldn't this fit here. I think it would have redundant to create another exception class like `NotLikedException` . For me how I see multipleLikeException is that handle the liking process. Maybe we could try to find a better name for this exception instead to incorporate multiple likes error and no likes.